[CI] SLMStatDisruptionIT testFailedSnapshotSubmitStats failing

**Build Scans:**
- [elasticsearch-periodic #12370 / openjdk21_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/vkq4nlgpudvai)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:slm:internalClusterTest" --tests "org.elasticsearch.xpack.slm.SLMStatDisruptionIT.testFailedSnapshotSubmitStats" -Dtests.seed=D731014A073536E1 -Dtests.locale=is -Dtests.timezone=America/Porto_Velho -Druntime.java=21
```

**Applicable branches:**
9.4

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.slm.SLMStatDisruptionIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testFailedSnapshotSubmitStats'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Master is not stable but the method expects a stable master node
```

**Issue Reasons:**
- [9.4] 1 consecutive failures in step openjdk21_checkpart2_java-matrix

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/slm/src/internalClusterTest/java/org/elasticsearch/xpack/slm/SLMStatDisruptionIT.java
+++ b/x-pack/plugin/slm/src/internalClusterTest/java/org/elasticsearch/xpack/slm/SLMStatDisruptionIT.java
@@ -575,6 +575,7 @@ public class SLMStatDisruptionIT extends AbstractSnapshotIntegTestCase {
             assertSnapshotPartial(repoName, snapshotName);
             assertMetadata(policyName, 0, 1, 1);
         }, 1, TimeUnit.MINUTES);
+        ensureStableCluster(2);
     }
 
     private void assertMetadata(String policyName, long taken, long failure, long invocationsSinceLastSuccess) {

